# 讀檔

In [1]:
import pandas as pd 
path = './112522/'
pd.set_option('display.max_rows', None)

all_episode = []
for episode in range(1,17):
    df = pd.read_csv(path + '112522_{}.csv'.format(episode))
    
    sec_10_interval = []
    for i in range(1, 144):
        f = ((i-1) * 100 <= df['time']) & (df['time'] < i * 100)
        sec_10_interval.append(len(df[f]))
    all_episode.append(sec_10_interval)

# 第6集 資料分析

In [2]:
df_count_6 = pd.DataFrame({
    'count':all_episode[5]
})
df_count_6[(df_count_6['count'] > 0)]

count
0       54
1       38
2       21
3       12
4       17
5       17
6        6
7        3
8        4
9        8
10       9
11       3
12       8
13       3
14       4
15      10
16       9
17      13
18      12
19       9
20      11
21      14
22       5
23       4
24       6
25      14
26      10
27       6
28      10
29      19
30      18
31      37
32      73
33      30
34      24
35      30
36      23
37      19
38      15
39      39
40      49
41      23
42      29
43      26
44      20
45      30
46      25
47      20
48      61
49      56
50      20
51      40
52      35
53      11
54      16
55      27
56      16
57      13
58      23
59      18
60      19
61      12
62      10
63      10
64      12
65      13
66      40
67      32
68      41
69      31
70       8
71      10
72       8
73      24
74      12
75      10
76       6
77      31
78      22
79      28
80      15
81      26
82      15
83      38
84      22
85       7
86      11
87       6
88       6
89      10
90      14
91      15
92      26
93      41
94      16
95      19
96      18
97      47
98       8
99      15
100     11
101     21
102     12
103      4
104     13
105     19
106     24
107     11
108     28
109     13
110     26
111     33
112     23
113     19
114     25
115     19
116      6
117     12
118     10
119      9
120      5
121     14
122     19
123     21
124     41
125     25
126     27
127     36
128     50
129     32
130     25
131     61
132     90
133     57
134     51
135     33
136     20
137     23
138     11
139      7
140     12
141     11
142      8

In [3]:
import numpy as np
print('Max', max(all_episode[5]))
print('Min', min(all_episode[5]))
print('Mean', np.mean(all_episode[5]))
print('Std', np.std(all_episode[5]))

Max 90
Min 3
Mean 20.95104895104895
Std 14.934977330105013


## 大於平均數的片段

In [4]:
f = df_count_6['count'] > np.mean(all_episode[5]) - 1
df_count_6[f]

count
0       54
1       38
2       21
31      37
32      73
33      30
34      24
35      30
36      23
39      39
40      49
41      23
42      29
43      26
44      20
45      30
46      25
47      20
48      61
49      56
50      20
51      40
52      35
55      27
58      23
66      40
67      32
68      41
69      31
73      24
77      31
78      22
79      28
81      26
83      38
84      22
92      26
93      41
97      47
101     21
106     24
108     28
110     26
111     33
112     23
114     25
123     21
124     41
125     25
126     27
127     36
128     50
129     32
130     25
131     61
132     90
133     57
134     51
135     33
136     20
137     23

## 大於兩個標準差片段

In [5]:
f = df_count_6['count'] > (np.mean(all_episode[5]) - 1) + (2 * np.std(all_episode[5]) - 1)
df_count_6[f]

count
0       54
32      73
40      49
48      61
49      56
128     50
131     61
132     90
133     57
134     51

0 簽到  
32 OP開始  
40 OP結束  
48 戰槌變身  
128 兵長登場  
131 顎要被砍了

# 第7集 資料分析

In [6]:
df_count_7 = pd.DataFrame({
    'count':all_episode[6]
})
df_count_7[(df_count_7['count'] > 0)]

count
0       58
1       29
2       46
3       26
4       13
5        6
6        4
7        6
8        6
9        5
10      30
11      17
12      20
13      18
14      28
15      39
16      64
17      30
18      11
19      20
20      17
21      21
22      18
23      74
24      32
25       9
26      21
27      12
28      11
29      13
30      12
31       5
32       5
33      12
34      10
35      11
36      11
37      14
38      17
39       5
40       6
41      10
42      12
43      12
44      12
45       9
46      19
47       7
48      10
49       8
50      10
51      20
52      14
53      12
54      15
55      37
56      16
57       9
58       9
59       6
60       5
61      11
62      16
63      28
64      65
65      22
66      21
67      17
68      10
69      27
70      21
71      38
72      30
73      31
74      14
75      14
76      26
77      19
78      14
79      30
80      33
81      51
82      29
83      24
84      21
85      30
86      13
87      21
88       8
89      11
90      11
91      13
92      15
93       3
94      14
95      17
96      20
97      18
98      38
99      17
100     12
101      8
102     11
103     20
104      6
105     21
106     13
107     38
108     24
109     16
110     30
111     15
112     18
113     38
114     18
115     35
116     48
117     36
118     20
119     29
120     35
121     50
122     19
123     33
124     11
125     26
126     26
127     44
128     33
129     23
130     22
131     32
132     80
133     38
134     23
135     21
136     10
137     19
138     14
139     10
140     12
141      8
142     19

In [7]:
import numpy as np
print('Max', max(all_episode[6]))
print('Min', min(all_episode[6]))
print('Mean', np.mean(all_episode[6]))
print('Std', np.std(all_episode[6]))

Max 80
Min 3
Mean 20.902097902097903
Std 13.893357413317753


## 大於平均數的片段

In [8]:
f = df_count_7['count'] > np.mean(all_episode[6]) - 1
df_count_7[f]

count
0       58
1       29
2       46
3       26
10      30
12      20
14      28
15      39
16      64
17      30
19      20
21      21
23      74
24      32
26      21
51      20
55      37
63      28
64      65
65      22
66      21
69      27
70      21
71      38
72      30
73      31
76      26
79      30
80      33
81      51
82      29
83      24
84      21
85      30
87      21
96      20
98      38
103     20
105     21
107     38
108     24
110     30
113     38
115     35
116     48
117     36
118     20
119     29
120     35
121     50
123     33
125     26
126     26
127     44
128     33
129     23
130     22
131     32
132     80
133     38
134     23
135     21

## 大於兩個標準差片段

In [9]:
f = df_count_7['count'] > (np.mean(all_episode[6]) - 1) + (2 * np.std(all_episode[6]) - 1)
df_count_7[f]

count
0       58
16      64
23      74
64      65
81      51
116     48
121     50
132     80

0 簽到  
16 OP開始  
23 OP結束  
64 阿爾敏炸彈  
81 皮克醬掉到地上  
116 胡桃鉗  
121 艾倫: 奴僕汁 讚!!!  
132 帥之巨人站起來了

# 第8集 資料分析

In [10]:
df_count_8 = pd.DataFrame({
    'count':all_episode[7]
})
df_count_8[(df_count_8['count'] > 0)]

count
0       44
1       26
2        5
3       14
4        5
5        6
6        8
7       11
8       12
9        5
10      28
11      66
12      32
13      25
14      25
15      25
16      18
17      15
18      18
19      60
20      16
21      13
22      12
23      17
24      13
25      16
26      22
27      11
28       9
29       7
30      10
31       6
32       4
33       4
34      10
35       7
36       8
37      13
38      21
39      14
40      13
41      14
42      13
43       8
44      10
45      17
46      13
47      19
48      17
49      20
50      17
51      18
52      12
53      33
54      19
55      34
56      25
57      29
58      14
59      17
60      39
61      22
62      37
63      29
64      23
65      13
66      12
67      33
68      37
69      15
70      18
71       9
72      24
73      28
74      21
75      40
76      25
77      12
78      20
79      20
80      36
81      63
82      98
83      24
84      48
85      21
86      23
87      33
88      19
89      22
90      24
91      30
92      26
93      23
94      14
95      12
96      11
97       8
98      11
99      11
100     22
101     26
102     10
103      8
104      6
105     13
106     14
107      7
108      3
109     18
110     12
111     11
112      9
113     15
114     11
115     14
116     16
117      8
118      4
119      5
120      8
121      4
122     10
123     36
124    117
125    105
126     73
127     46
128     31
129     26
130     33
131     25
132     19
133     25
134     25
135     18
136     20
137     12
138      7
139      9
140     11
141     17
142     11

In [11]:
import numpy as np
print('Max', max(all_episode[7]))
print('Min', min(all_episode[7]))
print('Mean', np.mean(all_episode[7]))
print('Std', np.std(all_episode[7]))

Max 117
Min 3
Mean 20.958041958041957
Std 17.61510561468257


## 大於平均數的片段

In [12]:
f = df_count_8['count'] > np.mean(all_episode[7]) - 1
df_count_8[f]

count
0       44
1       26
10      28
11      66
12      32
13      25
14      25
15      25
19      60
26      22
38      21
49      20
53      33
55      34
56      25
57      29
60      39
61      22
62      37
63      29
64      23
67      33
68      37
72      24
73      28
74      21
75      40
76      25
78      20
79      20
80      36
81      63
82      98
83      24
84      48
85      21
86      23
87      33
89      22
90      24
91      30
92      26
93      23
100     22
101     26
123     36
124    117
125    105
126     73
127     46
128     31
129     26
130     33
131     25
133     25
134     25
136     20

## 大於兩個標準差片段

In [13]:
f = df_count_8['count'] > (np.mean(all_episode[7]) - 1) + (2 * np.std(all_episode[7]) - 1)
df_count_8[f]

count
11      66
19      60
81      63
82      98
124    117
125    105
126     73

11 賈碧追上飛船 + OP開始  
19 OP結束  
81 莎夏被射到   
124 莎夏死了